src/Sarif.Viewer.VisualStudio.Core/Services/DataService.cs

```diff
@@ -48,34 +50,29 @@ public void SendEnhancedResultData(SarifLog sarifLog)
             SendEnhancedResultDataAsync(sarifLog).FileAndForget(Constants.FileAndForgetFaultEventNames.SendEnhancedData);
         }
 
-        private static async Task SendEnhancedResultDataAsync(SarifLog sarifLog)
+        private async Task SendEnhancedResultDataAsync(SarifLog sarifLog)
         {
             await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
             Assumes.NotNull(sarifLog);
             Assumes.True(sarifLog.Runs?.Count == 1);
 
-            var componentModel = (IComponentModel)AsyncPackage.GetGlobalService(typeof(SComponentModel));
-            if (componentModel != null)
-            {
-                var dte = Package.GetGlobalService(typeof(DTE)) as DTE2;
-                var projectNameCache = new ProjectNameCache(dte?.Solution);
-                var items = new List<SarifErrorListItem>();
-                Run run = sarifLog.Runs.First();
+            string logPath = $"{Guid.NewGuid()}.sarif";
+
+            await ErrorListService.ProcessSarifLogAsync(sarifLog, logPath, cleanErrors: false, openInEditor: false, monitorSarifFile: false);
 
-                foreach (Result r in run.Results)
-                {
-                    var sarifErrorListItem = new SarifErrorListItem(run, 0, r, string.Empty, projectNameCache);
-                    sarifErrorListItem.PopulateAdditionalPropertiesIfNot();
-                    items.Add(sarifErrorListItem);
-                }
+            SarifErrorListItem sarifItem = CodeAnalysisResultManager.Instance.CurrentRunDataCache.SarifErrors?[0];
+
+            if (sarifItem != null)
+            {
+                sarifItem.PopulateAdditionalPropertiesIfNot();
```

Comment:
>PopulateAdditionalPropertiesIfNot [](http://example.com/codeflow?start=26&length=33)

Does this get called for the other items in the error list?

Reply:
Current implementation these properties get populated when user navigates to (double click on) an error list item to postpone the loading time of these properties from loading Sarif content to navigating to specific item. 

If we expect to navigate to the item without manual operation, then yes we need to populate the properties for all item. Or we can add this call to navigate method so don't need to call it explicitly.